nomos-services/cryptarchia-consensus/src/mix/adapters/libp2p.rs

```diff
@@ -0,0 +1,86 @@
+use nomos_mix_service::{
+    backends::libp2p::{
+        Libp2pNetworkBackend, Libp2pNetworkBackendEvent, Libp2pNetworkBackendEventKind,
+        Libp2pNetworkBackendMessage,
+    },
+    NetworkMsg, NetworkService,
+};
+use overwatch_rs::services::{relay::OutboundRelay, ServiceData};
+use tokio::sync::broadcast::{self, error::RecvError};
+use tokio_stream::{wrappers::BroadcastStream, StreamExt};
+
+use crate::mix::{BoxedStream, NetworkAdapter};
+
+const BUFFER_SIZE: usize = 64;
+
+#[derive(Clone)]
+pub struct LibP2pAdapter {
+    network_relay: OutboundRelay<<NetworkService<Libp2pNetworkBackend> as ServiceData>::Message>,
+    mixed_msgs: broadcast::Sender<Vec<u8>>,
+}
+
+#[async_trait::async_trait]
+impl NetworkAdapter for LibP2pAdapter {
+    type Backend = Libp2pNetworkBackend;
+
+    async fn new(
+        network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
+    ) -> Self {
+        let relay = network_relay.clone();
+        let mixed_msgs = broadcast::Sender::new(BUFFER_SIZE);
+        let mixed_msgs_sender = mixed_msgs.clone();
+        // this wait seems to be helpful in some cases since we give the time
+        // to the network to establish connections before we start sending messages
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
Please add a todo here. We are planning to add a status system to await for services readyness, hopefully we can remove this kind of stuff.

Reply:
Added TODO 851c7af290fc44b51179552e5064357339b15b91